src/V3Stats.cpp

```diff
@@ -202,6 +243,42 @@ class StatsVisitor final : public VNVisitor {
         allNodes(nodep);
         iterateChildrenConst(nodep);
     }
+    virtual void visit(AstActive* nodep) override {
+        m_countRegs = m_isPreOrder && nodep->hasClocked();
+        if (m_countRegs) iterateChildrenConst(nodep);
+    }
+    virtual void visit(AstVarRef* nodep) override {
+        AstVar* varp = nodep->varp();
+        if (m_countRegs && nodep->access().isWriteOrRW() && varp->isSignal()
+            && !varp->isUsedLoopIdx()) {
+            AstVarScope* varScopep = nodep->varScopep();
+            if (varScopep->user1()) {  // Already seen at least one assignment to this var
+                // Only count assignments on _different_ source lines as different since array
+                // indexed assignments may have been expanded into multiple AstVarRef by now
+                if (nodep->fileline() != (FileLine*)varScopep->user2p()) {
+                    varScopep->user1(varScopep->user1() + 1);
+                    varScopep->user2p((AstNode*)nodep->fileline());
+                }
+            } else {
+                // A "Potential RAM" is rather arbitarily defined as any array signal:
+                // 1.) whose depth in elements is greater than its width in bits
+                // 2.) whose total number of bits exceeds the const threshold minRAMBits
+                // 3.) which is assigned in no more the 2 locations in the source
+                // This produces reasonable rersults in trials but could be improved
```

Comment:
Alternatively since you dump the row by column size, why not just not make this distinction and dump everything separately as if a RAM?

Reply:
I will revisit this. I'm considering mimicking the rules Xilinx FPGA synthesis uses to infer RAMs. From memory, I think this is something like A 2D array with a single indexed assignment in a clocked process, and a single indexed reference.
If I can make that work, I can dispense with the size based filtering altogether, and we should correlate well with FPGA synthesis. I'll check the inference rules ...

Reply:
Done, got rid of all 3 of these constraints